sdks/agent-sdk/src/core/Agent.ts

```diff
@@ -29,25 +29,57 @@ export interface AgentOptions<ContentTypes> {
   client: Client<ContentTypes>;
 }
 
-export type AgentEventHandler<ContentTypes = unknown> = (
+export type AgentMessageHandler<ContentTypes = unknown> = (
   ctx: AgentContext<ContentTypes>,
 ) => Promise<void> | void;
 
-export type AgentMiddleware<ContentTypes> = (
+export type AgentMiddleware<ContentTypes = unknown> = (
   ctx: AgentContext<ContentTypes>,
-  next: () => Promise<void>,
-) => Promise<void>;
+  next: () => Promise<void> | void,
```

Comment:
should this allow passing an error to `next` to active the error middlewares?

Reply:
We'll go simply with `next()` and `throw` but `return` should stop the middleware chain and do the final emit (currently not happening).